samples/matter/common/src/util/finite_dict.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <type_traits>
+
+namespace Nrf
+{
+/*
+   FiniteDict template container allows to wrap dictionaries between a specific K-type trivial key and T-type value.
+   The size or the container is predefined at compilation time, therefore
+   the maximum number of stored elements must be well known. FiniteDict owns
+   inserted values, meaning that once the user inserts a value it will not longer
+   be valid outside of the container, i.e. FiniteDict cannot return stored object by copy.
+   FiniteDict's API offers basic operations like:
+     * Inserting new values under provided key (Insert)
+     * Erasing existing items under a given key (Erase) - this operation does not take care of the
+	   duplicated items, meaning that the pointer under a given key is released unconditionally
+	   and the responsibility for the duplications is on the application side.
+     * Retrieving values stored under provided key ([] operator)
+     * Checking if the map contains a non-null value under given key (Contains)
+	 * Retrieving a number of free slots available in the map (FreeSlots)
+	Prerequisites:
+     * T must have move semantics and bool()/==operators implemented
+*/
+template <typename T, typename K, std::size_t N> struct FiniteDict {
```

Comment:
I'd rather refactor the existing FiniteMap to make it more versatile or  because now we end up with a lot of boiler plate code.

Reply:
Ok, I agree, apart from that a Map should translate from one typename to another, so we can extend the existing FiniteMap :)